tests/DCPS/XTypes/run_test.pl

```diff
@@ -23,7 +25,20 @@
   push(@common_args, "--verbose");
 }
 
-my %params = (
+my %params;
+if ($secure) {
+%params = (
+  "FinalStructMatch_s"        => {reader_type => "FinalStructSub", writer_type => "FinalStructPub", expect_to_fail => 0, reg_type => "PD_OL_OA_OM_OD", key_val => 2,
+                                r_ini => "rtps_disc_security.ini", w_ini => "rtps_disc_security.ini"},
+  "AppendableMatch_s"         => {reader_type => "AppendableStruct", writer_type => "AdditionalPostfixFieldStruct", expect_to_fail => 0, reg_type => "PD_OL_OA_OM_OD", key_val => 4,
+                                r_ini => "rtps_disc_security.ini", w_ini => "rtps_disc_security.ini"},
+  "AppendableNoMatch_s"       => {reader_type => "AppendableStruct", writer_type => "AdditionalPrefixFieldStruct", expect_to_fail => 1, reg_type => "PD_OL_OA_OM_OD", key_val => 5,
+                                r_ini => "rtps_disc_security.ini", w_ini => "rtps_disc_security.ini"},
+  "Dependency_s"              => {reader_type => "AppendableStruct", writer_type => "AppendableStructWithDependency", expect_to_fail => 0, reg_type => "PD_OL_OA_OM_OD", key_val => 14,
+                                r_ini => "rtps_disc_security.ini", w_ini => "rtps_disc_security.ini"},
+);
+} else {
```

Comment:
```suggestion
}
else {
```

For some reason this is the convention in perl.